lib/transproc/hash.rb

```diff
@@ -244,7 +244,9 @@ def nest!(hash, root, keys)
 
       if nest_keys.size > 0
         child = Hash[nest_keys.zip(nest_keys.map { |key| hash.delete(key) })]
-        hash.update(root => child)
+        old_nest = hash[root]
+        new_nest = old_nest.is_a?(Hash) ? old_nest.merge(child) : child
+        hash.update(root => new_nest)
```

Comment:
This wouldn't work if `hash.fetch(root, {})` returns not a hash.

This would break [the second spec for the `nest`](https://github.com/solnic/transproc/pull/47/files#diff-7278ebb7e7df672b615966a9c93991c6R186)

This is why I define `old_nest` and check it in line 248